src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
In that case I would opt for dropping the dependency and going for the low tech option I suggested earlier. The reason I say this is once the dependency is baked in, it's API usage would only increase over time proliferating throughout the code base. This should not be allowed to happen if there is an imminent solution. I can also only find this in it's current form from people that forked the cli ages ago on github. The current master branch of the cli does not appear to have this code anymore. There be dragons here. 

@alinapopa - Can you help out with this discussion? Where is this project currently maintained? Where has it moved to?